api/types.go

```diff
@@ -313,7 +303,7 @@ func DefaultOptions() Options {
 		MirostatEta:      0.1,
 		PenalizeNewline:  true,
 
-		NumThread: runtime.NumCPU(),
+		NumThread: 0, // let the runtime decide
```

Comment:
llama.cpp uses some more in-depth logic to pick an appropriate number of threads, so let the runner handle it